iroh-resolver/src/unixfs_builder.rs

```diff
@@ -198,44 +197,14 @@ impl File {
     }
 
     pub fn encode(self) -> impl Stream<Item = Result<(Cid, Bytes)>> {
-        async_stream::try_stream! {
-            let mut cids = Vec::new();
-            let nodes = self.nodes;
-            tokio::pin!(nodes);
-
-            while let Some(node) = nodes.next().await {
-                let node = node?;
-                let bytes = node.encode()?;
-                let cid = Cid::new_v1(Codec::Raw as _, cid::multihash::Code::Sha2_256.digest(&bytes));
-                cids.push((cid, bytes.len()));
-                yield (cid, bytes);
-            }
-
-            if cids.len() > 1  {
-                // yield root as last element, as we now have all links
-                let links = cids.into_iter().map(|(cid, len)| {
-                    dag_pb::PbLink {
-                        hash: Some(cid.to_bytes()),
-                        name: Some("".into()),
-                        tsize: Some(len as u64),
-                    }
-                }).collect();
-
-                let inner = unixfs_pb::Data {
-                    r#type: DataType::File as i32,
-                    ..Default::default()
-                };
-                let outer = encode_unixfs_pb(&inner, links)?;
-                let node = UnixfsNode::Directory(Node { outer, inner });
-                let bytes = node.encode()?;
-                let cid = Cid::new_v1(Codec::DagPb as _, cid::multihash::Code::Sha2_256.digest(&bytes));
-                yield (cid, bytes)
-            }
-        }
+        TreeBuilder::balanced_tree().stream_tree(self.nodes)
```

Comment:
lets store a `tree_builder`, which is initialized to balanced_tree by default on the builder and use that here